simple.zig

```diff
@@ -0,0 +1,42 @@
+const std = @import("std");
+const WordMap = std.StringHashMap(u32);
+
+pub fn main() !void {
+    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+    const ally = &arena.allocator;
+
+    var words = WordMap.init(ally);
+    const stdin = std.io.bufferedReader(std.io.getStdIn().reader()).reader();
+    var buf: [1024]u8 = undefined;
+    while (try stdin.readUntilDelimiterOrEof(&buf, '\n')) |line| {
+        for (line) |*c| c.* = std.ascii.toLower(c.*);
+        var it = std.mem.tokenize(line, " ");
+        while (it.next()) |word| {
+            const ret = try words.getOrPut(word);
```

Comment:
Interesting. I presume `getOrPut` is to avoid hashing the key twice?

Reply:
Yep that's right. In hindsight I probably should have called the variable `entry` here instead of `ret`.